src/components/AllReviews.jsx

```diff
@@ -1,29 +1,57 @@
 import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from "@mui/material"
 import { useEffect, useState } from "react"
-import { getAllReviews } from "./api"
+import { getAllReviews, getCategories } from "./api"
 import ReviewRow from "./reviewRow"
+import { useNavigate, useParams } from "react-router-dom"
 
 function AllReviews(){
     const [allReviews, setAllReviews] = useState([])
     const [isLoading, setIsLoading] = useState(true)
+    const { category } = useParams()
+    const [reviewParams, setReviewParams] = useState({})
+    const [reviewErr, setReviewErr] = useState(null)
+    const [allCategories, setAllCategories] = useState([])
+    const navigate = useNavigate()
 
+    const CategoryHandler = (event) => {
+        event.preventDefault()
+        navigate(`/reviews/${event.target.value}`, {replace: true})
+        setReviewParams((currentParams)=>{
+            return {...currentParams, category: event.target.value}
+        })
+    }
     
     useEffect(()=>{
         setIsLoading(true)
-        getAllReviews()
+        getCategories()
+        .then((categoriesArr)=>{
+            setAllCategories(categoriesArr)
+        })
+        getAllReviews({category})
         .then((reviews)=>{
             setAllReviews(reviews)
             setIsLoading(false)
         })
-    },[])
+        .catch((err)=>{
+            setReviewErr(err.response.data.msg)
+            setIsLoading(false)
+        })
+    },[reviewParams, category])
+
     return isLoading ? (<p>Loading...</p>):(
         <TableContainer>
+            {reviewErr?<p><br/>{reviewErr}</p>:
             <Table>
                 <TableHead>
                     <TableRow>
                         <TableCell>ID #</TableCell>
                         <TableCell>Title</TableCell>
-                        <TableCell>Category</TableCell>
+                        <TableCell>
+                            <select name="category" id="category" onClick={CategoryHandler}>
+                                <option key='default' value="">Categories</option>
```

Comment:
This might have to be an onChange instead of onClick, currently as soon as I click the drop down the whole page re-renders and the drop down is deselected, I can't actually manage to select a category.